qcfractal/interface/client.py

```diff
@@ -82,6 +83,27 @@ def __init__(self,
 
         self.server_name = self.server_info["name"]
 
+        def version_list(version):
+            version_match = re.search(r"\d+\.\d+\.\d+", version)
+            if version_match is None:
+                raise ValueError(f"Could not read version of form XX.YY.ZZ in {version}")
+            version = version_match.group(0)
+            return [int(x) for x in version.split(".")]
+
+        from . import __version__  # Import here to avoid circular import from __init__
+        try:
+            server_version_min_client = version_list(self.server_info["client_lower_version_limit"])
+            server_version_max_client = version_list(self.server_info["client_upper_version_limit"])
+        except KeyError:
+            raise IOError(f"The Server at {self.address}, version {self.server_info['version']} does not report what "
+                          f"Client versions it accepts! It can be almost asserted your Client is too new for the Server "
+                          f"you are connecting to.")
+        client_version = version_list(__version__)
+        assert server_version_min_client <= client_version <= server_version_max_client, (
+                f"This Client of version {client_version} does not fall within the Server's allowed Client versions "
+                f"of [{server_version_min_client}, {server_version_max_client}]. Server address: {self.address}"
```

Comment:
Perhaps note that it is likely that they need to upgrade their portal version with `pip install qcportal` or `conda install qcportal -c conda-forge.`


These kinds of error messages both here and above should tell a user what to do to fix the issue not just tell them there is an issue.